doc/releases/changelog-dev.md

```diff
@@ -208,6 +208,9 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The `MeasurementRegister` class is introduce so that native-mcm workflows return tuples of measurement processes, not an _ad hoc_ dictionary of mid-circuit measurements.
+  [(#5461)](https://github.com/PennyLaneAI/pennylane/pull/5461)
+
```

Comment:
Curious if this is just a technical improvement, or whether users might notice any difference?

Reply:
@albi3ro Correct me if I'm wrong, while this should not change user interface, users might notice a difference due to the absence of bug in the long run. Not sure if that counts.